### DISCUSSION TITLE
Why can an ordinary object respond to change?

### DISCUSSION THREAD

**Discussion by cnsource** [vuejs/pinia#1404] [Help and Questions] [answered]
### Reproduction

https://stackblitz.com/edit/github-cjfmgk-bpwdd4?file=src/App.vue

### Steps to reproduce the bug

B is a normal variable, but it can respond to changes

### Expected behavior

B is added to the array and will no longer have a response

### Actual behavior

B is added to the array and still has a response

### Additional information

You can view the console log

**Comment by posva** [comment, marked as answer]
Because it's a Ref, you need to create a copy of the object with `{...object.value}`

**Reply by cnsource** [reply]
thank you

**Reply by cnsource** [reply]
![image](https://user-images.githubusercontent.com/22205879/176055081-46df940a-4c20-4a0e-adf1-ef2f4bc2ac63.png)
A and B look the same when printed on the console. B uses toRaw to get the original data. I don't know why B is still a responsive variable after toRaw

I think that B gets the reference of the original data and stores it in the array, but the student response variable will modify the original data, which will cause B to be changed. Am I right?


This is my way to prove the above conjecture
```javascript
<script setup lang="ts">
import {ref, toRaw} from "vue";

let A = {a:10};
let B = ref(A);
let C = toRaw(B.value)
function changeData(){
  B.value.a += 20
  console.log(A,B,C)
}
</script>

<template>
  <hr/>
  <button @click="changeData">changeData</button>
  <br>
  A:{{ A }}<br>
  B:{{ C }}<br>
  C:{{ B }}<br>
</template>

```
![image](https://user-images.githubusercontent.com/22205879/176057485-a726b83d-db7f-42c3-a8e1-8035116ff45a.png)

**Reply by posva** [reply]
Even if you use toRaw, you are still modifying the same underlying object
